src/Main.elm

```diff
@@ -52,51 +52,129 @@ type DecimalStatus
 executeAdd : Model -> Model
 executeAdd model =
     { model
-        | display = String.fromFloat <| model.valueA + model.valueB
+        | display = showResults <| model.valueA + model.valueB
         , valueA = 0
-        , actionA = Blank
+        , actionA = Nothing
         , valueB = 0
-        , actionB = Blank
+        , actionB = Nothing
         , displayStatus = ShowingResults
     }
 
 
+executeSubtract : Model -> Model
+executeSubtract model =
+    { model
+        | display = String.fromFloat <| model.valueA - model.valueB
+        , valueA = 0
+        , actionA = Nothing
+        , valueB = 0
+        , actionB = Nothing
+        , displayStatus = ShowingResults
+    }
+
+
+executeMultiply : Model -> Model
+executeMultiply model =
+    { model
+        | display = String.fromFloat <| model.valueA * model.valueB
+        , valueA = 0
+        , actionA = Nothing
+        , valueB = 0
+        , actionB = Nothing
+        , displayStatus = ShowingResults
+    }
+
+
+executeDivide : Model -> Model
+executeDivide model =
+    { model
+        | display = String.fromFloat <| model.valueA / model.valueB
+        , valueA = 0
+        , actionA = Nothing
+        , valueB = 0
+        , actionB = Nothing
+        , displayStatus = ShowingResults
+    }
+
+
+showResults : Float -> String
+showResults float =
+    let
+        string =
+            String.fromFloat float
+
+        hasDecimal =
+            String.any (\a -> '.' == a) string
+
+        largerThenTen =
+            case hasDecimal of
+                True ->
+                    String.length string > 11
+
+                False ->
+                    String.length string > 10
+    in
+    case ( string, largerThenTen ) of
+        ( text, False ) ->
+            text
+
+        ( _, True ) ->
+            "ERROR"
+
+
 
 -- Update
 
 
 type Msg
     = UpdateDisplay CalButton
     | Saveinput CalButton
+    | AllClear
 
 
-buttonToAction : CalButton -> Action
+buttonToAction : CalButton -> Maybe Action
 buttonToAction button =
     case button of
         CalAction action ->
-            action
+            Just action
 
         _ ->
-            Blank
+            Nothing
+
+
+savedValueBs : Model -> CalButton -> Model
+savedValueBs model button =
+    { model
+        | valueB = displatToFloat model.display
+        , actionB = buttonToAction button
+        , displayStatus = ShowingResults
+    }
 
 
 saveAction : Model -> CalButton -> Model
 saveAction model button =
-    case ( model.actionA, model.actionB ) of
-        ( Blank, Blank ) ->
+    case model.actionA of
+        Nothing ->
```

Comment:
It's kind of weird that `saveAction` takes a `CalButton` instead of an `Action`.

Reply:
Now using Action type directly and not using Calbutton to get to 'Action'